src/store/audio.ts

```diff
@@ -1687,124 +1656,50 @@ export const audioStore = createPartialStore<AudioStoreTypes>({
     ),
   },
 
-  PLAY_AUDIO: {
-    action: createUILockAction(
-      async ({ commit, dispatch }, { audioKey }: { audioKey: AudioKey }) => {
-        const audioElem = audioElements[audioKey];
-        audioElem.pause();
-
-        // 音声用意
-        let blob = await dispatch("GET_AUDIO_CACHE", { audioKey });
-        if (!blob) {
+  FETCH_AUDIO: {
+    async action({ commit, dispatch }, { audioKey }: { audioKey: AudioKey }) {
```

Comment:
この関数はやってることがfetchだけじゃなく、キャッシュ検索や、何を生成中かを制御する機能を持っているのでそれがわかるように（あるいはせめて勘違いしないように）名前を変えてあげるのがよさそうに感じました！

･･････`PLEPARE_AUDIO`？
```suggestion
  /**
   * キャッシュがなければ音声を生成して返す。
   */
  PLEPARE_AUDIO: {
    async action({ commit, dispatch }, { audioKey }: { audioKey: AudioKey }) {
```

Reply:
そもそも生成中かの制御が`GENERATE_AUDIO`内でされていないのが問題のように見えたので、その制御を`GENERATE_AUDIO`側に移しました。
これなら個人的なイメージだとFETCHそのものなんですが、もしかしたら一般的には違うかもです…`FIND_CACHE_OR_GENERATE_AUDIO`とかのほうが良いでしょうか？

Reply:
あ、なるほどです！
JavaScript文脈でのfetchというとネットからダウンロードする系のやつのイメージがあって、実際に音声をエンジンで生成してダウンロードするAPIもあるので、エンジンを直接叩く関数に勘違いしちゃうかも。
どういうのと、fetchの割には機能が多い（ほぼgetのみの機能をイメージしますが、プログレスバーの表示機能がある）かなと！

Reply:
何かいい考え方ないかなと思って色々調べてみました。
この[Qiita記事](https://qiita.com/alt_yamamoto/items/25eda376e6b947208996#star5%E9%81%A9%E5%88%87%E3%81%AA%E5%90%8D%E5%89%8D%E3%82%92%E3%81%A4%E3%81%91%E3%82%8B)がちょっと近いかもです。

> 名前をつけるときは、以下を心がける。
> ・他人が見ても名前から目的や意味がわかる
> ・実際の用途や処理と名前が合っている
> ・名前に書かれている以外の処理や役目を持たない

最後の「名前に書かれている以外の処理や役目を持たない」が近そうかもです。

Reply:
直接叩くのと区別がつかないのはなるほどです。`FETCH`は使わないほうが良さそうですね。
プログレスバーの表示機能の方は以下にまとめて書きました。
https://github.com/VOICEVOX/voicevox/pull/1492#discussion_r1297631148